### PR TITLE
Feat: Invoice Search Field

### LINKED ISSUE #468: Campo: Búsqueda > Factura Venta/Compra
Soporte al formulario de búsqueda de Factura como campo

###  _**Versión ZK:**_

https://user-images.githubusercontent.com/45974454/196731841-1e5de380-4505-43ea-8a25-397ca9f46eb4.mp4

### _**Versión Vue:**_

https://user-images.githubusercontent.com/45974454/196731927-717881fc-5096-433d-84d5-20619511c10e.mp4

### PR DESCRIPTION
**Pull request by elsiosanchez**
#### Screenshot or Gif

https://github.com/solop-develop/frontend-core/assets/45974454/d5b67a1f-2666-4078-9a5a-f6a9a9e4f361

#### Issues

Fixed https://github.com/solop-develop/frontend-core/issues/468

### CODE REVIEW

File: src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue [hunk 1]
```diff
@@ -18,14 +18,13 @@
...
-      style="margin: 0px; width: 100%"
-      @visible-change="showList"
-      @change="currentValue()"
+      clearable
+      remote
+      :loading="isLoading"
+      :remote-method="remoteMethod"
+      @visible-change="showOrder"
     >
       <el-option
```
Comment (EdwinBetanc0urt) [inline src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue:33]: please add `EmptyOption` component before `el-option` list.

Comment (EdwinBetanc0urt) [reply src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue:33]: Manage as https://github.com/solop-develop/frontend-core/blob/feature/legacy-support/src/components/ADempiere/FieldDefinition/FieldSearch/ProductInfo/PanelForm/QueryCriteria/productCategoryField.vue#L38-L39 to improve peformance

File: src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue [hunk 2]
```diff
@@ -64,59 +63,83 @@ export default defineComponent({
...
         })
-    }
-
-    const currentValue = () => {
-      store.dispatch('searchInvociesInfos', {
-        order_id: saleOrderField.value
-      })
+          .then(response => {
+            const { records } = response
+            optionsListOrder.value = records.map((list) => {
```
Comment (EdwinBetanc0urt) [inline src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue]: Remove loop to improve peformance https://github.com/solop-develop/frontend-core/blob/feature/legacy-support/src/components/ADempiere/FieldDefinition/FieldSearch/ProductInfo/PanelForm/QueryCriteria/productCategoryField.vue#L119

**Review by EdwinBetanc0urt — CHANGES_REQUESTED** [review]

File: src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue (hunk 1 above)
Comment (elsiosanchez) [reply src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue:33]: https://github.com/solop-develop/frontend-core/pull/2220/commits/64508a3abac781fa837483fd62e7f96c800c9b13

File: src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue (hunk 2 above)
Comment (elsiosanchez) [reply src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/OrderField.vue]: https://github.com/solop-develop/frontend-core/pull/2220/commits/64508a3abac781fa837483fd62e7f96c800c9b13

File: src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/businessPartner.vue [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    >
+      <empty-option-select
+        :current-value="currentSaleOrder"
+        :is-allows-zero="false"
+      />
+      <el-option
+        v-for="(option, key) in optionsList"
+        :key="key"
+        :value="option.id"
+        :label="option.displayColumn"
```
Comment (EdwinBetanc0urt) [inline src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/businessPartner.vue]: Implement as `:label="item.values.DisplayColumn"` to improve peformance

File: src/store/modules/ADempiere/field/search/invoice.js [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+    invoiceBusinessPartnerList({ commit }, searchValue) {
+      return new Promise(resolve => {
+        let list
+        requestListBusinessPartners({
+          search_value: searchValue,
+          page_size: 300
+        })
+          .then(response => {
+            const { records } = response
+            list = records.map((list) => {
```
Comment (EdwinBetanc0urt) [inline src/store/modules/ADempiere/field/search/invoice.js]: Remove this loop to improve peformance.

**Review by EdwinBetanc0urt — CHANGES_REQUESTED** [review]

File: src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/businessPartner.vue (hunk 3 above)
Comment (elsiosanchez) [reply src/components/ADempiere/FieldDefinition/FieldSearch/Invoice/PanelForm/QueryCriteria/businessPartner.vue]: https://github.com/solop-develop/frontend-core/pull/2220/commits/9531be9e2962182759a5db75cd377bcda4bb6ece

File: src/store/modules/ADempiere/field/search/invoice.js (hunk 4 above)
Comment (elsiosanchez) [reply src/store/modules/ADempiere/field/search/invoice.js]: https://github.com/solop-develop/frontend-core/pull/2220/commits/9531be9e2962182759a5db75cd377bcda4bb6ece

**Review by EdwinBetanc0urt — APPROVED** [review]

**Pull request merged**